projects/firebaseui-angular-library/src/lib/dynamic-loader.service.ts

```diff
@@ -0,0 +1,113 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable } from '@angular/core';
+
+export interface Resource {
+    name: string;
+    type: "css" | "js";
+    src: string;
+}
+
+export interface LoadedResource {
+    [name: string]: {
+        loaded: boolean,
+        type: "css" | "js",
+        src: string
+    }
+}
+
+@Injectable()
+export class DynamicLoaderService {
+
+    private static ResourcesStore: Resource[] = [];
+    private static LoadedResources: LoadedResource = {};
+
+    private _document: Document | undefined = undefined;
+
+    constructor(@Inject(DOCUMENT) _document?: any) {
+        this._document = _document;
+
+        DynamicLoaderService.ResourcesStore.forEach((res) => {
+            DynamicLoaderService.LoadedResources[res.name] = {
+                loaded: false,
+                type: res.type,
+                src: res.src
+            };
+        });
+    }
+
+    /**
+     * Loads a series of previously registered Script(s)
+     * @param resNames The list of resources to load
+     */
+    load(...resNames: string[]) {
+        let promises: any[] = [];
+        promises = resNames.map((name) => this.loadResource(name));
```

Comment:
This can be written as a shorthand form. No need for initializing the array first. Additionally, the array should be correctly typed - not with `any`.